### DISCUSSION TITLE
SaasError: $string is not a string

### DISCUSSION THREAD

**Discussion by FlorianRuen** [webpack/webpack#16719] [Q&A]
Hello there,

Just upgraded some packages in my Spring project, including boottstrap, and now my project return an error during compilation.
I try to search many solutions across the web, but didn't find something that suits me

```
ERROR in ./src/main/webapp/app/app.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/postcss-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[1].use[3]!./src/main/webapp/app/app.scss)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: $string: #ff6932 is not a string.
    ╷
132 │   @if str-index($string, "data:image/svg+xml") {
    │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
  node_modules/bootstrap/scss/_functions.scss 132:7               escape-svg()
  node_modules/bootstrap/scss/_navbar.scss 23:41                  @import
  node_modules/bootstrap/scss/bootstrap.scss 27:9                 @import
  src/main/webapp/app/scss/0.broth/_bootsrap-variables.scss 56:9  @import
  src/main/webapp/app/scss/global.scss 4:9                        @import
  src/main/webapp/app/app.scss 2:9                                root stylesheet
```

My file _bootsrap-variables.scss contains only variable to override default bootstrap colors such as 

```
$primary: #ff6932;
$secondary: #3b7794;
$success: #1aae9f;
$info: #3b7794;
$warning: #efa544;
$danger: #d3455b;
$minor: #a1a7ac;
```

Here is the versions I'm using for bootstrap, saas-loader and webpack

- "webpack": "^5.65.0",
-  "webpack-cli": "^4.10.0",
- "webpack-dev-server": "^4.7.1",
- "webpack-merge": "5.8.0",
- "webpack-notifier": "1.15.0",
- "postcss-loader": "7.0.1"
- "node-sass": "^7.0.3",
- "bootstrap": "5.2.0",

Any idead on how I can fix this error to compile my project ?

Kindly,
Florian

**Comment by alexander-akait** [comment]
Looks like you have invalid type of `$string` variable, please read the error message or ask for the help in bootstrap/sass repo
